rust/core-lib/src/view.rs

```diff
@@ -980,13 +1005,30 @@ impl View {
     /// indicates a search for the next occurrence past the end of the file.
     pub fn select_next_occurrence(&mut self, text: &Rope, reverse: bool, wrapped: bool,
                                   _allow_same: bool, modify_selection: &SelectionModifier) {
+        let (cur_start, cur_end) = match self.selection.last() {
+            Some(sel) => (sel.min(), sel.max()),
+            _ => (0, 0)
+        };
+
         // multiple queries; select closest occurrence
         let closest_occurrence = self.find.iter().flat_map(|x|
             x.next_occurrence(text, reverse, wrapped, &self.selection)
         ).min_by_key(|x| {
             match reverse {
-                true => x.end,
-                false => x.start
+                true => {
```

Comment:
I might choose to write this,

```rust
match reverse {
    true if x.end > cur.start => ...,
    true => ...,
    false if x.start < cur_end => ...,
    false => ...,
}
```

with the same caveat as before, but maybe more so, i.e. I suspect more people disagree with me on this one. :)

Reply:
This looks for sure cleaner 👍